core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/SdkInternalExecutionAttribute.java

```diff
@@ -34,11 +35,23 @@
 
     /**
      * The key to store the {@link CompletableFuture} returned by {@link AsyncResponseTransformer#prepare()} method
-     * in the first attempt of a request. This is used only for async streaming requests
+     * in the first attempt of a request. This used to be used only for async streaming requests.
+     *
+     * The purpose this attribute used to service has been replaced with
+     * {@link SdkInternalExecutionAttribute#EXECUTION_ATTEMPT} which now tracks the execution attempt number and is
+     * incremented with successive retries.
      */
+    @Deprecated
```

Comment:
Sucks we marked SdkInternalExecutionAttribute as protected

Reply:
I agree. Resolving this conversation because there's nothing you or I can do about it at this point.